### PR TITLE
Fix CellBlock3D::getCellBoundingBox()

### PR DESCRIPTION
**Pull request by FG-TUM**
# Description

Fixes the value clamping in `CellBlock3D::getCellBoundingBox()` when we calculate the bounding box of a cell at the lower edge of the domain.

- [x] Fixed bug
- [x] Adapted some tangentially related documentation.

## Additional Context

- This caused the particle duplication bug in ls1 because particles were not found by the `LeavingParticleCollector`. Thus during the update, some halos were not removed and their corresponding owned partners were placed right on top of them.
- *Why was this not an issue before?* Not sure. I think when I rewrote the Iterators in #712 I started using `CellBlock3D::getCellBoundingBox()` significantly more than it was used previously. Later when I rewrote `LeavingParticleCollector` in #792 this was used even more, especially around domain borders. So maybe we just didn't use the function for these edge cases before? Seems like another important argument for introducing test coverage -> #807 

# How Has This Been Tested?

- [x] fixed bug in ls1
- [x] New unit tests

### CODE REVIEW

File: src/autopas/containers/CellBlock3D.h [hunk 1]
```diff
@@ -447,16 +447,18 @@ inline std::pair<std::array<double, 3>, std::array<double, 3>> CellBlock3D<Parti
...
-      boxmax[d] = this->_cellLength[d];
     } else if (index3d[d] == _cellsPerInteractionLength) {
+      // Case: we are at the lower boundary of the non-halo box
       boxmin[d] = _boxMin[d];
     }
-    // no else, as this might ALSO be 1
+
+    boxmax[d] = boxmin[d] + this->_cellLength[d];
+
+    // This is not an else to the if block above, if there is only one cell
```
Comment (SamNewcome) [inline src/autopas/containers/CellBlock3D.h:461]: Can you rephrase this? I'm not super sure what is meant.

```suggestion
    // This is not necessarily an else case to the if block above e.g. in the case that there is only one cell
```

**Review by SamNewcome — CHANGES_REQUESTED** [review]

**Review by thesamriel — APPROVED** [review]

File: src/autopas/containers/CellBlock3D.h (hunk 1 above)
Comment (FG-TUM) [reply src/autopas/containers/CellBlock3D.h]: How about that: 4288da1e2

File: src/autopas/containers/CellBlock3D.h [hunk 2]
```diff
@@ -447,16 +447,19 @@ inline std::pair<std::array<double, 3>, std::array<double, 3>> CellBlock3D<Parti
...
     } else if (index3d[d] == _cellsPerInteractionLength) {
+      // Case: we are at the lower boundary of the non-halo box
       boxmin[d] = _boxMin[d];
     }
-    // no else, as this might ALSO be 1
+
+    boxmax[d] = boxmin[d] + this->_cellLength[d];
+
+    // This must not be an else to the if block above.
+    // If the above is true, this can also be true if there is e.g. only one cell
```
Comment (SamNewcome) [inline src/autopas/containers/CellBlock3D.h:462]:
```suggestion
    // This must not be an else to the if block above as both cases can be true.
    // e.g. if there is only one cell
```

**Review by SamNewcome — APPROVED** [review]
